src/main/resources/com/ncc/aif/restricted_hypothesis_aif.shacl

```diff
@@ -51,3 +52,41 @@ aida:HypothesisEntityClusterRequireHandle
         sh:minCount 1 ;
     ] .
 
+########################
+# 4.3 TA3 #1.a All KEs in model should be referenced by hypothesis
+#------------------------
+aida:KEsMustBeReferenced
+    a sh:SPARQLConstraint ;
+    sh:message "This KE is unreferenced by the hypothesis" ;
+    sh:select """
+        PREFIX aida:  <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/InterchangeOntology#>
+        SELECT $this
+        WHERE {
+            ?hypothesis a aida:Hypothesis .
+            FILTER NOT EXISTS { ?hypothesis aida:hypothesisContent/aida:subgraphContains $this }
+        }
+    """
+    .
+
+aida:AllKEsReferencedShape
+    a sh:NodeShape ;
+    sh:targetClass aida:Entity, aida:Event, aida:Relation ;
+    sh:target [
+        a sh:SPARQLTarget ;
+        sh:select """
+            PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+            PREFIX rdfs:  <http://www.w3.org/2000/01/rdf-schema#>
+            PREFIX aidaDomainCommon: <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/AidaDomainOntologiesCommon#>
+            SELECT ?argumentAssertion
+            WHERE {
+                ?argumentAssertion a rdf:Statement .
+                ?argumentAssertion rdf:predicate ?pred .
+                { ?pred rdfs:subClassOf+|rdf:type/rdfs:subClassOf* aidaDomainCommon:RelationArgumentType }
```

Comment:
Not for the LDCOntologies because we control how those are created. However, if someone wanted to use a different ontology that was a concrete instance of subclass of RelationArgumentType, this second path would pick it up.

Reply:
As we discussed, these restrictions are only ever used by NIST so we probably do not have to worry about anyone using a different ontology. If we are really concerned with optimization we may want to take it out. I am not sure if there is a cost for the extra property path lookup.